### PR TITLE
fix: remove ServiceAccount binding to the edit role

### PR DESCRIPTION
**Pull request by VanillaSpoon**
# Issue link
closes https://github.com/project-codeflare/codeflare-operator/issues/161

# What changes have been made
In the current setup, we have a clusterrolebinding between the ServiceAccount and the default edit ClusterRole. The edit role provides an expansive set of permissions which have been checked and deemed not necessary for the operators functionality. 

Therefor this PR removes the clusterrolebinding. 

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by sutaakar** [comment]
@VanillaSpoon There is an error logged in operator pod:
```
E1003 15:07:58.460889       1 queuejob_controller_ex.go:2097] Failed to delete resources associated with app wrapper: 'test-ns-tcbrs/mnist', err 1 error occurred:
	* jobs.batch is forbidden: User "system:serviceaccount:openshift-operators:codeflare-operator-controller-manager" cannot list resource "jobs" in API group "batch" at the cluster scope
```
That is probably the reason for the test failure.

**Comment by astefanutti** [comment]
> @VanillaSpoon There is an error logged in operator pod:
> 
> ```
> E1003 15:07:58.460889       1 queuejob_controller_ex.go:2097] Failed to delete resources associated with app wrapper: 'test-ns-tcbrs/mnist', err 1 error occurred:
> 	* jobs.batch is forbidden: User "system:serviceaccount:openshift-operators:codeflare-operator-controller-manager" cannot list resource "jobs" in API group "batch" at the cluster scope
> ```
> 
> That is probably the reason for the test failure.

@VanillaSpoon @sutaakar I'd suggest we add permissions for batch Job resources to the default operator role.

**Comment by VanillaSpoon** [comment]
I appreciate the support on this @astefanutti and @sutaakar. 

The changes have been updated in 9762026b9d817cd4e5d22218b2d57dfdd499104e

**Review by sutaakar — COMMENTED** [review]
/lgtm

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -18,6 +18,14 @@ rules:
...
   - update
   - watch
+- apiGroups:
+  - batch
+  resources:
+  - jobs
+  verbs:
+  - list
+  - delete
+  - create
```
Comment (astefanutti) [inline config/rbac/role.yaml:28]: While that works, MCAD should theoretically also be able to update these resources, so I'd suggest to add the following verbs for consistency:

```
- patch
- update
- watch
```

Comment (VanillaSpoon) [reply config/rbac/role.yaml:26]: Thanks @astefanutti  :) I have applied the update in aa18a415df6598f91d20f785dc64a967c140a411

**Comment by astefanutti** [comment]
Thanks a lot @VanillaSpoon, that's a critical fix w.r.t. security.

**Pull request merged**